servo/connectors/kubernetes.py

```diff
@@ -1892,13 +1898,13 @@ async def raise_for_status(self) -> None:
 
         # Check for failure conditions
         self._check_conditions(status.conditions)
-        await self.raise_for_pod_status()
+        await self.raise_for_pod_status(adjustments=adjustments)
 
         # Catchall
         self.logger.trace(f"unable to map deployment status to exception. Deployment: {self.obj}")
         raise RuntimeError(f"Unknown Deployment status for '{self.name}': {status}")
 
-    async def raise_for_pod_status(self):
+    async def raise_for_pod_status(self, adjustments: List[servo.Adjustment]):
```

Comment:
I think we may want to rename the method to be more semantically aligned since we're sorta beyond pod status at this point.

Maybe something like `raise_for_failed_pod_adjustments` or something?